test/plugin/base_test.rb

```diff
@@ -2718,11 +2728,30 @@ def assert_prometheus_metric_value(metric_name, expected_value, labels = {})
                      # Sum up all metric values regardless of the labels.
                      metric.values.values.reduce(0.0, :+)
                    else
-                     metric.get(labels: labels)
+                     metric.get(labels)
                    end
     assert_equal(expected_value, metric_value)
   end
 
+  def assert_opencensus_metric_value(metric_name, expected_value, labels = {})
+    metric_name = Monitoring::OpenCensusMonitoringRegistry
+                  .translate_metric_name(metric_name)
+    labels = labels.transform_keys(&:to_s).transform_values(&:to_s)
+    stats_recorder = OpenCensus::Stats.ensure_recorder
+    view_data = stats_recorder.view_data metric_name
+    assert_not_nil(view_data)
+    # For now assume all metrics are counters.
+    assert_kind_of(OpenCensus::Stats::Aggregation::Sum,
+                   view_data.view.aggregation)
+    assert_true(view_data.view.measure.int64?)
+    tag_values = view_data.view.columns.map { |column| labels[column] }
```

Comment:
Dumb question:

Is my understanding correct that the order of the tag keys in `view_data.view.columns` are deterministic? The [implementation](https://github.com/census-instrumentation/opencensus-ruby/blob/091a7fd5eba1879f71a928e5ba3d29e8d1446de3/lib/opencensus/stats/view.rb#L27) / official doc did not mention anything regarding to that. But based on the fact the array can be used as key to [OpenCensus::Stats::ViewData::data](https://github.com/census-instrumentation/opencensus-ruby/blob/091a7fd5eba1879f71a928e5ba3d29e8d1446de3/lib/opencensus/stats/view_data.rb#L17), it must be, right?

It was a bit confusing initially to see `tag_values` used as a key later (line 2750). After digging through the OpenCensus code, it makes sense though.

Reply:
Are you asking because you see a problem? I borrowed that idiom from the OpenCensus tests.

I wouldn't assume too much about the status of this particular OpenCensus implementation, given that it's only half finished. On the upside, implementation details such as that are likely to be binding, and I've seen the same field order assumptions in views in OpenCensus implementations for other languages.

Reply:
I was wondering if the test might be flaky if the order is not deterministic. Let's get this in and address that if it indeed becomes an issue then.